[BE][PG NCCL] Improve input mismatch error msg

rohan-varma opened the pull request:
Test Plan: CI

Differential Revision: D48363238

rohan-varma commented on torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp:
```diff
@@ -1502,8 +1502,16 @@ std::vector<at::Tensor> flatten_for_scatter_gather(
+              " with world_size",
+              world_size,
+              " and ",
+              num_devices,
```
this is always 1 afaik, so can also exclude it.

awgu commented on torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp:
```diff
@@ -1502,8 +1502,16 @@ std::vector<at::Tensor> flatten_for_scatter_gather(
+              "but got ",
+              tensor_lists[i].size(),
+              " inputs",
+              " with world_size",
```
nit: I think?
```suggestion
              " with world_size ",
```

awgu reviewed (approved):
LGTM! Double check the nit.

rohan-varma committed:
[BE] Improve input mismatch error msg (#107281)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/107281

Test Plan: CI

Reviewed By: daniellepintz, H-Huang, awgu

Differential Revision: D48363238

fbshipit-source-id: 705bc6cc63d4173fba630f69471fbfa0d35bb636

rohan-varma commented:
@pytorchbot merge -f "CI done"
